File: Android/java/app/src/main/java/org/oreilly/nmd/MainActivity.java
package org.oreilly.nmd;

import android.app.Activity;
import android.content.Intent;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;

public class MainActivity extends Activity {

  private CheckBox mTermsCheckbox;
  private Button mBrowseButton;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    mTermsCheckbox = findViewById(R.id.terms_checkbox);
    mTermsCheckbox.setOnCheckedChangeListener(this::onCheckChanged);
    mBrowseButton = findViewById(R.id.browse_button);
    mBrowseButton.setOnClickListener(this::browseContent);
    SQLiteDatabase database = new DbHelper(this, "db", null, 1).getWritableDatabase();
    database.close();
  }

  private void onCheckChanged(Button button, boolean checked) {
    mBrowseButton.setEnabled(checked);
  }

  private void browseContent(View view) {
    Intent intent = new Intent(this, BrowseContentActivity.class);
    startActivity(intent);
  }

}

File: Android/kotlin/app/src/main/java/org/oreilly/nmd/MainActivity.kt
package org.oreilly.nmd

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        browse_button.setOnClickListener(::browseContent)
        terms_checkbox.setOnCheckedChangeListener(::onCheckChanged)
    }

    private fun browseContent(view: View) {
        val intent = Intent(this, BrowseContentActivity::class.java)
        startActivity(intent)
    }

    private fun onCheckChanged(button: Button, checked: Boolean) {
        browse_button.isEnabled = checked
    }

}